rcl/test/rcl/test_publisher.cpp

```diff
@@ -490,3 +535,294 @@ TEST_F(CLASSNAME(TestPublisherFixture, RMW_IMPLEMENTATION), test_invalid_publish
     rcl_publish_serialized_message(nullptr, &serialized_msg, nullptr));
   rcl_reset_error();
 }
+
+// Mocking rmw_publisher_count_matched_subscriptions to make
+// rcl_publisher_get_subscription_count fail
+TEST_F(CLASSNAME(TestPublisherFixtureInit, RMW_IMPLEMENTATION), test_mock_publisher_count) {
```

Comment:
@Blast545 nit: I think it'd be nice if test names would correlate with what it is being tested and not so much the fact that a mock is used. WDYT?

Reply:
It would be good if we reach a consensus, I think I named the first half with the function mocked and the other with the one tested. I think we can agree to something like "test_mock_function_tested". If you think that's ok I'll proceed renaming.

Reply:
Improved the naming with 24a0941